### PR TITLE
EPPT-3458 fsi mandatory ffmc clipping

### PR DESCRIPTION
**Pull request by ryan-cocking-mo**
[Ticket](https://metoffice.atlassian.net/browse/EPPT-3458)

Ensures fine fuel moisture content is always clipped to values in the range 0 - 101. 

This prevents NaN values infrequently occurring in the initial spread index.

### CODE REVIEW

**Review by katherinetomkins-mo — DISMISSED** [review]

**Review by robertplatt-mo — APPROVED** [review]

**Review by robertplatt-mo — CHANGES_REQUESTED** [review]
I approved and then had a rethink. Is is possible to write a test for the scenario described in EPPT-3458. This is quite the edge case. If you see my comment on the original task https://metoffice.atlassian.net/browse/EPPT-3121?focusedCommentId=669463 I struggled to create a scenario where the clipping was ever needed. Having found one incorporating it in to a test would avoid any future confusion about whether the clip is required (as was present in original task to add clipping https://metoffice.atlassian.net/browse/EPPT-3121)

File: improver/fire_weather/initial_spread_index.py [hunk 1]
```diff
@@ -96,7 +96,7 @@ def _calculate_spread_factor(self) -> np.ndarray:
         spread_factor = (
             91.9
             * np.exp(self.moisture_content * -0.1386)
-            * (1.0 + (self.moisture_content**5.31) / 4.93e7)
+            * (1.0 + (np.abs(self.moisture_content) ** 5.31) / 4.93e7)
```
Comment (robertplatt-mo) [inline improver/fire_weather/initial_spread_index.py]: The ticket mentions self.moisture_content being negative in rare cases because the clip option wasn't in use. If that's correct then 324 should fix all negative values. Adding np.abs here surely removes the possibility of any negative values therefore bringing in to question whether 324 has actually fixed the underlying issue. What have I missed?

File: improver_tests/fire_weather/test_fine_fuel_moisture_code.py [hunk 2]
```diff
@@ -620,6 +599,8 @@ def test_moisture_conversion_with_clip(moisture_content_val, expected_array_val)
         (10.0, 15.0, 95.0, 5.0, 85.0, 20.70),
         # Case 4: Precipitation just below threshold (should not adjust)
         (20.0, 0.4, 50.0, 10.0, 85.0, 86.82),
+        # Case 5: Extreme values should should be clipped to maximum (edge case)
```
Comment (robertplatt-mo) [inline improver_tests/fire_weather/test_fine_fuel_moisture_code.py]: typo
```suggestion
        # Case 5: Extreme values should be clipped to maximum (edge case)
```

Comment (robertplatt-mo) [reply improver_tests/fire_weather/test_fine_fuel_moisture_code.py]: Or slight rephrase instead 

```suggestion
        # Case 5: Extreme values must be clipped to between 0 and 101 to avoid negative numbers in the initial_spread_index calculations
```

**Comment by robertplatt-mo** [comment]
New test added. Tests fails when the unclipped ffmc is returned instead of the clipped. Ideally I was after a test in Initial Spread Index as it was there that we found the nans but happy to accept as is.

**Review by robertplatt-mo — DISMISSED** [review]

File: improver_tests/fire_weather/test_fine_fuel_moisture_code.py (hunk 2 above)
Comment (ryan-cocking-mo) [reply improver_tests/fire_weather/test_fine_fuel_moisture_code.py]: Rephrased comment with reference to negative moisture content, which is unphysical

**Review by robertplatt-mo — APPROVED** [review]
